File: Noodle/noodle/node/interfaces.cpp
#include "interfaces.hpp"
#include "noodle/macros.hpp"
namespace Noodle {
EMSCRIPTEN_BINDINGS(Noodle) {
    EM_OBJECT(NodePin)
        .field("name", &NodePin::name)
        .field("type", &NodePin::type)
        .field("direction", &NodePin::direction)
        .field("metadata", &NodePin::metadata)
    ;

    EM_OBJECT(PinPath)
        .field("nodeId", &PinPath::nodeId)
        .field("pinName", &PinPath::pinName)
    ;

    EM_OBJECT(NodeConnection)
        .field("id", &NodeConnection::id)
        .field("from", &NodeConnection::from)
        .field("to", &NodeConnection::to)
    ;
}
}

File: web/src/lib/Bluefoot/Noodle/ctypes/Interfaces.ts
import { NoodleMetadata } from "./Metadata";

export type NodeID = string;

export interface NodePin {
    name : string;
    type : string;
    direction : string;
    metadata : NoodleMetadata;
}

export interface PinPath {
    nodeId : NodeID;
    pinName : string;
}

export interface NodeConnection {
    id : number;
    from : PinPath;
    to : PinPath;
}